### DISCUSSION TITLE
Getting errors when using CopilotChat when switching a session

### DISCUSSION THREAD

**Discussion by gisketch** [CopilotC-Nvim/CopilotChat.nvim#119] [Q&A] [answered]
![CleanShot 2024-03-05 at 10 31 38](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/78424395/a043c4ca-c639-4edf-8a8e-1faa8a0c79fb)

I used explain and chat buffer here and I got these errors. Weird because it worked the first time, and then when I tried in another project it stopped working. Im getting an invalid buffer id.

Edit: confirmed this error occurs when switching a session, the first session on startup works, but upon switching to another, the error now occurs and chat becomes unusable.

**Comment by deathbeam** [comment, marked as answer]
Can you try the same thing on the canary branch? I think it should be already fixed there but not 100% sure, if it isnt its probably relatively easy fix tho
